src/main/java/io/cryostat/net/web/http/api/beta/graph/labels/SetMatcher.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright The Cryostat Authors
+ *
+ * The Universal Permissive License (UPL), Version 1.0
+ *
+ * Subject to the condition set forth below, permission is hereby granted to any
+ * person obtaining a copy of this software, associated documentation and/or data
+ * (collectively the "Software"), free of charge and under any and all copyright
+ * rights in the Software, and any and all patent rights owned or freely
+ * licensable by each licensor hereunder covering either (i) the unmodified
+ * Software as contributed to or provided by such licensor, or (ii) the Larger
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.beta.graph.labels;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public class SetMatcher implements LabelMatcher {
+
+    private final SetMatcher.Operator operator;
+    private final String key;
+    private final Set<String> values;
+
+    SetMatcher(String key, SetMatcher.Operator operator) {
+        this(key, operator, Set.of());
+    }
+
+    SetMatcher(String key, SetMatcher.Operator operator, Collection<String> values) {
+        this.key = key;
+        this.operator = operator;
+        this.values = new HashSet<>(values);
+    }
+
+    @Override
+    public String getKey() {
+        return key;
+    }
+
+    @Override
+    public boolean test(String s) {
+        return operator.with(values).test(s);
+    }
+
+    public enum Operator {
+        IN("In", args -> v -> contains(args, v)),
+        NOT_IN("NotIn", args -> v -> !contains(args, v)),
```

Comment:
How do I test an expression with more than one value in the set? When I tried querying with `descendantTargets (filter: { annotations: "PORT in (9091, 9096)" })`, the results only show the `cryostat:9091` container:
```
query {
    environmentNodes(filter: { name: "JDP" }) {
        name
        nodeType
        descendantTargets (filter: { annotations: "PORT in (9091, 9096)" }){
            target {
                alias
                annotations {
                cryostat
                }
            }
        }
    }
}
```
```
{
    "data": {
        "environmentNodes": [
            {
                "descendantTargets": [
                    {
                        "target": {
                            "alias": "io.cryostat.Cryostat",
                            "annotations": {
                                "cryostat": {
                                    "HOST": "cryostat",
                                    "JAVA_MAIN": "io.cryostat.Cryostat",
                                    "PORT": "9091"
                                }
                            }
                        }
                    }
                ],
                "name": "JDP",
                "nodeType": "Realm"
            }
        ]
    }
}
```

Reply:
Ah, darn. Thanks for catching that. This is actually a bug due to the way that label selectors can be `AND`ed together by joining them with a comma, but the set membership expression also uses the comma inside the parentheses to denote the set members.

So it's supposed to be a supported syntax to do say `PORT in (9091, 9096), env != production`, but my naive approach to "parsing" breaks down here because of the comma and splits the expression inappropriately.

I see two solutions. The first is perhaps "most correct" but also the most annoying, and that's to implement a more proper parser rather than just leaning on `string.split(",")` and regexes, so that we can parse the expression and separate out the "clauses" joined by commas, and then parse out the different parts of the expressions as well.

The second is to deviate slightly from the k8s label selector syntax and just not do clause conjunction using a comma, but instead split that out into a list of clauses. The GraphQL query would change from `filter: { annotations: "PORT in (9091, 9096), env != production" }` to `filter: { annotations: ["PORT in (9091, 9096)", "env != production"] }. This is quick and easy, and IMO even somewhat nicer because we lean on the GraphQL syntax that provides some notion of a type system, rather than just encoding everything into one big string.

Reply:
Unresolving this - we can do a query like `filter: { annotations: "PORT in (9091, 9096)" }` now, but we're now missing the ability to `AND` multiple annotation conditions like that together within the query, which is what ex. `filter: { annotations: "PORT in (9091, 9096), env != production" }` was supposed to give us. I can either implement support for that syntax myself, or I can change the syntax a little and let GraphQL's parser do the work by making it into a list type in the query filter object.

Reply:
>  let GraphQL's parser do the work by making it into a list type in the query filter object.

I think it makes sense to use a list here. Seems like a good middle ground between the GraphQL syntax and the k8s syntax

Reply:
@ebaron Any opposition to using a GraphQL list type?

Reply:
That's pretty much my opinion on it, too. It might be a minor tripping point for a Kubernetes user who sees the familiar label selector syntax and tries to just copy-paste a selector conjunction they're already using elsewhere, but that's the only potential downside I see and it's an easy enough fix once they realize it's not working and check what our expected syntax actually is.